Enhances `Makefile` to make build dependencies self documenting.

wilmoore opened the pull request:
### Summary
- All build tasks requiring `npm` dependencies depend on `node_modules`
  target.
- `node_modules` file/directory target depends on `package.json` and
  `dist/*/*.min.js` files. Person building can never forget a step.
- Use the variable `$(RM)`.
### Benefit

No need to remember to run `npm install` or `git submodule ...` when a new component submodule is added or `package.json` is updated. `make` file dependency tracking takes care of this.
### Acceptance Criteria

```
% cd /tmp
% rm -rf brick ; git clone https://github.com/mozilla/brick.git ; cd brick
% make demos # notice the error
% curl -s https://raw.github.com/wilmoore/brick/quick-contributor-start/Makefile > Makefile
% make demos # this time all the dependencies will be worked out and the demos will be created
% make demos # all up-to-date tasks won't run again this time...the demos will be made either way since it sits in a `.PHONE` target
```

The above steps should result in updated submodules, npm install, and demos being created without errors.

wilmoore committed:
Enhances `Makefile` to make build dependencies self documenting.

- All build tasks requiring `npm` dependencies depend on `node_modules`
  target.
- `node_modules` file/directory target depends on `package.json` and
  `dist/*/*.min.js` files. Person building can never forget a step.
- Use the variable `$(RM)`.

ldoubleuz commented:
Hi Wil, is there a particular reason why `dist/*/*.min.js` is included as a requirement for `node_modules`? These dist files are generated by `build/minify.js`, so while this works because the dist files are already included in the repository, it would probably be best to avoid the circular dependency.

wilmoore committed:
correct submodule dependency for node_modules => component/**/component.json

wilmoore committed:
extract submodule dependency into component task

wilmoore committed:
Merge branch 'master' of https://github.com/mozilla/brick into quick-contributor-start

wilmoore commented:
> is there a particular reason why dist/_/_.min.js is included as a requirement for node_modules

Actually, I got this all wrong the first time. I've updated the PR with better separation of concerns and I've updated the `Acceptance Criteria` so it is easier to test.

The main value here is that if you add a new component or one of the submodules is broken, the `make` dependency rules should force a `git submodule update --init --recursive`.

Hopefully, when someone clones this project, they should just be able to run `make demos` to get a sense that everything is in order before they start contributing.
